src/witmotion_ros.cpp

```diff
@@ -127,17 +131,7 @@ ROSWitmotionSensorController::ROSWitmotionSensorController()
           .get_parameter_value()
           .get<bool>();
 
-  if (imu_enable_accel) {
-
-    node->declare_parameter(
-        "imu_publisher.measurements.acceleration.covariance",
-        std::vector<double>({-1, 0, 0, 0, 0, 0, 0, 0, 0}));
-    imu_accel_covariance =
-        node->get_parameter(
-                "imu_publisher.measurements.acceleration.covariance")
-            .get_parameter_value()
-            .get<std::vector<double>>();
-  }
+  load_parameter(imu_enable_accel, "imu_publisher.measurements.acceleration.covariance", -1.0, imu_accel_covariance);
```

Comment:
General question for all these replacements: are the initial values still handled? It is better to hold them known during initialisation

Reply:
In case they are not provided in the parameters yaml file, yes. Also, using -1 is useful as other modules (such as ekf pose) recognice negative covariances as a "do not use"  value.